daemons/ipa-kdb/ipa_kdb_mspac.c

```diff
@@ -2165,7 +2166,14 @@ krb5_error_code ipadb_sign_authdata(krb5_context context,
                                   "currently not supported.");
     }
 
-    if (is_as_req && with_pac && (flags & KRB5_KDB_FLAG_INCLUDE_PAC)) {
+    /* we need to create a PAC if we are requested one and this is an AS REQ,
+     * or we are doing protocol transition (s4u2self) */
+    if ((is_as_req && (flags & KRB5_KDB_FLAG_INCLUDE_PAC)) ||
+        (flags & KRB5_KDB_FLAG_PROTOCOL_TRANSITION)) {
```

Comment:
I think if should be `(flags & KRB5_KDB_FLAGS_S4U)` because both protocol transition and constraint delegation are in `KRB5_KDB_FLAGS_S4U` and `fetch_kdb_authdata()` does check against this constant when looking if it is included.